subiquity/models/subiquity.py

```diff
@@ -146,6 +147,13 @@ def set_source_variant(self, variant):
                 self._confirmation_task.cancel()
         else:
             self._install_event.set()
+        unconfigured_postinstall_model_names = \
+            self._cur_postinstall_model_names - self._configured_names
+        if unconfigured_postinstall_model_names:
+            if self._postinstall_event.is_set():
+                self._postinstall_event = asyncio.Event()
+        else:
+            self._postinstall_event.set()
```

Comment:
Conceptually speaking, this new code should only run if self._confirmation is set. I guess  `SystemSetupModel.__init__` should just set its _confirmation event as soon as it is created.

Reply:
In general... all this code currently falls in a kind of uncanny valley between being generic and specific. It allows for complete freedom in model names -- which is good for WSL -- but it assumes that all these model names fall into an install / confirm / postinstall flow -- which is not remotely appropriate for WSL. It feels like we could extract a base class from SubiquityModel that defers to subclasses the definition of this flow somehow, and then SystemSetupModel could inherit from this base class and not SubiquityModel. But this has flow on implications for the things like the values the ApplicationState enum can take so ehhhhh. Yet another thing to try to clean up after merge.

Reply:
Ack. This is a longer term goal.